src/menu.rs

```diff
@@ -77,80 +75,76 @@ pub fn context_menu<'a>(
         .into()
     };
 
-    let mut selected_dir = 0;
-    let mut selected = 0;
-    tab.items_opt().map(|items| {
-        for item in items.iter() {
-            if item.selected {
-                selected += 1;
-                if item.metadata.is_dir() {
-                    selected_dir += 1;
-                }
-            }
-        }
-    });
+    let selected = tab.selection_counts();
 
-    let mut children: Vec<Element<_>> = Vec::new();
+    let mut children: Vec<Element<_>> = Vec::with_capacity(16);
     match tab.location {
         Location::Path(_) | Location::Search(_, _) => {
-            if selected > 0 {
-                if selected_dir == 1 && selected == 1 || selected_dir == 0 {
+            if selected.any() {
+                if selected.exactly_one_dir() && selected.exactly_one() || selected.no_dirs() {
                     children.push(menu_item(fl!("open"), Action::Open).into());
                 }
-                if selected == 1 {
+                if selected.exactly_one() {
                     children.push(menu_item(fl!("open-with"), Action::OpenWith).into());
-                    if selected_dir == 1 {
+                    if selected.exactly_one_dir() {
                         children
                             .push(menu_item(fl!("open-in-terminal"), Action::OpenTerminal).into());
                     }
                 }
                 // All selected items are directories
-                if selected == selected_dir {
+                if selected.only_directories() {
                     children.push(menu_item(fl!("open-in-new-tab"), Action::OpenInNewTab).into());
                     children
                         .push(menu_item(fl!("open-in-new-window"), Action::OpenInNewWindow).into());
                 }
-                children.push(container(horizontal_rule(1)).padding([0, 8]).into());
-                children.push(menu_item(fl!("rename"), Action::Rename).into());
-                children.push(menu_item(fl!("cut"), Action::Cut).into());
-                children.push(menu_item(fl!("copy"), Action::Copy).into());
+                children.extend(vec![
```

Comment:
The `vec!` macro is not needed here, plain arrays implement `IntoIterator` as well. 

Also, wouldn't the use of `extend` versus a series of pushes make it less convenient to conditionally add menu items, like (currently pending) PRs like #359 and #370 do?

I guess I don't see the benefit of using `extend`. If it is to reduce the number of re-allocations as the `children` vec grows, this should already be solved by initializing with `with_capacity`.

Reply:
Personally, for what the code does now this would be too much abstraction. After all, right now all the builder would do is add elements to a vector. I don't know what the "minimum number of builder methods to warrant a builder pattern" would be, but I think it's more than 2. 

Also if I think about other places where builders are used, it's usually places where the creation is a complex process that needs to preserve certain invariants for every possible construction, and where different users create differently built objects. But here, there would only be one call to `build()`, because this is a very internal use, compared to builders that are part of the public API of HTTP libraries, database query builders, command line parsers, etc...

If this was in libcosmic, this should definitely be a builder pattern.

But ultimately, this is a decision someone from the pop-os team should make.